src/preamble.js

```diff
@@ -1004,7 +1006,7 @@ function enlargeMemory() {
 #else
   // TOTAL_MEMORY is the current size of the actual array, and DYNAMICTOP is the new top.
 #if ASSERTIONS
-  assert(DYNAMICTOP >= TOTAL_MEMORY);
+  assert(HEAPU32[DYNAMICTOP_PTR>>2] >= TOTAL_MEMORY);
```

Comment:
Btw, here, shouldn't `DYNAMICTOP` need to be strictly larger than `TOTAL_MEMORY`? That is, the address pointed to by dynamic top itself itself is not allocated, so `HEAPU32[HEAPU32[DYNAMICTOP_PTR>>2]]` is not valid indexable memory? If `DYNAMICTOP` == `TOTAL_MEMORY`, that should be a good condition that does not need enlarging?